### PR TITLE
handle queue creation when tags are different

### PR DESCRIPTION
**Pull request by itajaja**

### CODE REVIEW

File: tqp/topic_queue_poller.py [hunk 1]
```diff
@@ -17,27 +17,43 @@ def _jsonify_dictionary(dictionary):
...
-            tags={"tqp": "true", "dlq": "false", **tags, **extra_tags},
+    def _create_queue():
+        return sqs_resource.create_queue(
+            QueueName=name, Attributes=attributes, tags=tags,
         )
 
-    dead_letter_queue = _create_queue(
+    try:
+        return _create_queue()
+    except sqs_client.exceptions.QueueNameExists:
```
Comment (taion) [inline tqp/topic_queue_poller.py:34]: so this will only get hit if the existing tags are different?

File: tqp/topic_queue_poller.py [hunk 2]
```diff
@@ -17,27 +17,43 @@ def _jsonify_dictionary(dictionary):
...
+        queue_url = sqs_resource.get_queue_by_name(QueueName=name).url
+        existing_tags = sqs_client.list_queue_tags(QueueUrl=queue_url)["Tags"]
+        tags_to_remove = list(set(existing_tags.keys()) - set(tags.keys()))
+
+        sqs_client.tag_queue(QueueUrl=queue_url, Tags=tags)
+        if tags_to_remove:
+            sqs_client.untag_queue(QueueUrl=queue_url, TagKeys=tags_to_remove)
```
Comment (taion) [inline tqp/topic_queue_poller.py:41]: might we not also need to update values on existing tags?

**Comment by itajaja** [comment]
> why not try to fetch the queue by name first,

that doesn't give me much because I want to upsert anyway to make sure the attributes are in sync

File: tqp/topic_queue_poller.py (hunk 2 above)
Comment (itajaja) [reply tqp/topic_queue_poller.py:41]: `tag_queue` at line 39 does that

File: tqp/topic_queue_poller.py (hunk 1 above)
Comment (itajaja) [reply tqp/topic_queue_poller.py:34]: no, also if attributes are different. I don't want to parse the text of the error message though, so I'd rather adjust the tags and retry

Comment (taion) [reply tqp/topic_queue_poller.py:34]: okay, i'm just saying – it won't get hit every time?

Comment (itajaja) [reply tqp/topic_queue_poller.py:34]: nope

**Pull request merged**